arangod/RestServer/DumpLimitsFeature.cpp

```diff
@@ -0,0 +1,147 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#include "DumpLimitsFeature.h"
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/PhysicalMemory.h"
+#include "Basics/application-exit.h"
+#include "Logger/LogMacros.h"
+#include "ProgramOptions/Parameters.h"
+#include "ProgramOptions/ProgramOptions.h"
+#include "ProgramOptions/Section.h"
+
+using namespace arangodb;
+using namespace arangodb::application_features;
+using namespace arangodb::options;
+
+namespace {
+uint64_t defaultMemoryUsage() {
+  if (PhysicalMemory::getValue() >= (static_cast<uint64_t>(4) << 30)) {
+    // if we have at least 4GB of RAM, the default size is (RAM - 2GB) * 0.2
+    return static_cast<uint64_t>(
+        (PhysicalMemory::getValue() - (static_cast<uint64_t>(2) << 30)) * 0.2);
+  }
+  if (PhysicalMemory::getValue() >= (static_cast<uint64_t>(2) << 30)) {
+    // if we have at least 2GB of RAM, the default size is 256MB
+    return (static_cast<uint64_t>(256) << 20);
+  }
+  // for everything else the default size is 64MB
+  return (static_cast<uint64_t>(64) << 20);
```

Comment:
Why do we need 3 cases? I understand two: one larger than 4gb, another is smaller.
But is it common that we between 2 and 4 gb of memory?

Reply:
Removed one of the cases.